File: ipc/c/signal/windows/sender.c
#include <windows.h>
#include <stdio.h>

int main() {
    HANDLE hEvent = OpenEvent(EVENT_MODIFY_STATE, FALSE, L"MyEvent");
    if (hEvent == NULL) {
        fprintf(stderr, "Error opening event (%d)\n", GetLastError());
        return 1;
    }
    printf("Sending signal...\n");
    SetEvent(hEvent);
    CloseHandle(hEvent);
    return 0;
}

File: ipc/python/signal/windows/sender.py
import win32event
import win32api
import pywintypes

EVENT_NAME = "Global\\MyEvent"

def event_sender():
    try:
        event_handle = win32event.OpenEvent(win32event.EVENT_MODIFY_STATE, False, EVENT_NAME)
    except pywintypes.error as e:
        print(f"Error: Unable to open event ({e})")
        return

    print("Signaling event...")
    win32event.SetEvent(event_handle)

    win32api.CloseHandle(event_handle)

if __name__ == "__main__":
    input("Press Enter to send event... ")
    event_sender()
